geopandas/geodataframe.py

```diff
@@ -646,6 +658,14 @@ def __getitem__(self, key):
             result.__class__ = DataFrame
         return result
 
+    def __setitem__(self, key, value):
+        if key == self._geometry_column_name:
+            try:
+                value = _ensure_geometry(value, crs=self.crs)
```

Comment:
Should `self.crs` only be passed if `value` itself has no CRS? (is that case tested?)